src/gennylib/include/gennylib/ExecutionStrategy.hpp

```diff
@@ -99,34 +99,15 @@ class ExecutionStrategy {
         }
     }
 
-    // This function takes the existing counter metrics recorded in this class, and adds them into
-    // the metrics namespace timeseries. This may be desired in specific edge cases or at the end of
-    // each phase. This will always be called when the ExecutionStrategy is dtor'd.
-    void recordMetrics();
```

Comment:
I'm not sure ExecutionStrategy should have raii semantics - it should have the same lifecycle as the Actor itself; if there's still anything pending being recorded during actor dtor then I think there's something seriously wrong.

Reply:
Ehhh, so I agree, but removing cheap safeguards because they're not necessary now is usually the start of warning stories. Again, really doesn't matter here but I felt the need to put a soapbox under my feet again.

Reply:
> removing cheap safeguards because they're not necessary now is usually the start of warning stories

It took me a while to reason about why the scope-guard was necessary here and how it could be re-worked with the changed API fwiw. So counter-argument is that treating something with RAII semantics that doesn't really "need it" for correctness could cause confusion and incorrect usage in the future.